### PR TITLE
Fix yaml definitions

### PR DESCRIPTION
**Pull request by scisco**
**Related Issue(s):** #
N/A

**Proposed Changes:**

1. Updates the open api schemas to update the `next` field definition and match the definitions in the markdown with the ones given in the yaml file

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

**Comment by hgs-truthe01** [comment]
I'm a little confused on this one. I know we said we didn't want page to be part of the spec, but I didn't think we were adding next as a query string parameter? I thought it was intended to link to the next page using whatever parameter the implementer decided on and was returned in the search results. next as a parameter doesn't make a tone of sense to me. Do we need to specify it's format?

**Comment by scisco** [comment]
@hgs-truthe01 I think `next` is only needed as a query parameter when you using the `GET /stac/search`. On `POST /stac/search` you will pass it in the body.

Nevertheless, my PR is only updating the yaml file to reflect what is already added here: https://github.com/radiantearth/stac-spec/blob/dev/api-spec/api-spec.md#filter-parameters-and-fields

File: api-spec/openapi/STAC.yaml [hunk 1]
```diff
@@ -116,100 +116,60 @@ components:
...
+      name: next
       in: query
-      description: |
-        The optional page parameter returns the specified page of results
-        (with each page having size=limit).
-
-        * Minimum = 1
-        * Default = 1
+      description:
+        $ref: '#/components/schemas/next/description'
```
Comment (m-mohr) [inline api-spec/openapi/STAC.yaml]: I guess this works in most openAPI implementations, but it is not allowed by openAPI. The description objects (and there are more similar references here) don't allow $refs to be used. Only strings are allowed.

Comment (m-mohr) [reply api-spec/openapi/STAC.yaml]: The OpenAPI spec is not JSON Schema and works differently. $refs are not allowed by default, but must be allowed explicitly. See for example the [Components object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#componentsObject) where it explicitly states that Reference Objects (= $refs) are allowed. The descriptions are only of type string and don't mention Reference Object, therefore $refs are not allowed for descriptions.

Comment (scisco) [reply api-spec/openapi/STAC.yaml]: @m-mohr updated my PR. Could you take a look again?

**Pull request merged**